File: contract/src/main/java/rrlane/leasing/contract/service/CustomerService.java
package rrlane.leasing.contract.service;

import rrlane.leasing.contract.entity.Customer;

import java.util.Date;

public interface CustomerService {
    public void saveCustomer(Customer customer);
    public Customer viewCustomerByName(String name);
    public Customer viewCustomerByNameAndBirthdate(String name, Date birthDate);

}

File: contractapi/src/main/kotlin/de/leasing/contract/api/service/CustomerService.kt
package de.leasing.contract.api.service

import de.leasing.contract.api.entity.dto.CustomerReq
import de.leasing.contract.api.entity.dto.CustomerResp
import de.leasing.contract.api.entity.dto.LeasingContractResp
import java.util.*

interface CustomerService {
    fun saveCustomerRecord(customerReq: CustomerReq): CustomerResp
    fun listAllCustomerRecords(page: Int, size: Int): Iterable<CustomerResp>
    fun listAllCustomerRecords(sortBy: String?): Iterable<CustomerResp>
    fun getCustomerById(id: UUID): CustomerResp
}
